geth/node/node.go

```
@@ -163,6 +170,27 @@ func activatePersonalService(stack *node.Node, config *params.NodeConfig) error
 	})
 }
 
+func activateStatusService(stack *node.Node, config *params.NodeConfig) error {
+	if !config.StatusServiceEnabled {
+		logger.Info("Status service api is disabled")
+		return nil
+	}
+
+	if config.WhisperConfig == nil || !config.WhisperConfig.Enabled {
```

I am not sure this is needed. If one wants to activate StatusService and Whisper is not found, it should throw an error.